### DISCUSSION TITLE
Supabase Storage Move Method 400 Error

### DISCUSSION THREAD

**Discussion by BloxsmithDevelopment** [supabase/supabase#8637] [Questions]
Hello! We are trying to simply move a file using the `move()` method but are running into the following error:
![Screen Shot 2022-08-24 at 8 50 32 PM](https://user-images.githubusercontent.com/105516235/186549315-0bd94568-f9df-492c-9f4e-22468e551b66.png)

We have the follow RLS policies added to the bucket and that is it:
```
drop policy if exists "Public Access Bucket Product Images Select" on storage.objects;
create policy "Public Access Bucket Product Images Select"
on storage.objects for select
using ( bucket_id = 'product-media' );

drop policy if exists "Public Access Bucket Product Images Upload" on storage.objects;
create policy "Public Access Bucket Product Images Upload"
on storage.objects for insert
WITH CHECK ( bucket_id = 'product-media' );

drop policy if exists "Public Access Bucket Product Images Update" on storage.objects;
create policy "Public Access Bucket Product Images Update"
on storage.objects for update
WITH CHECK ( bucket_id = 'product-media' );

drop policy if exists "Public Access Bucket Product Images Delete" on storage.objects;
create policy "Public Access Bucket Product Images Delete"
on storage.objects for delete
using ( bucket_id = 'product-media' and auth.uid() = owner );
```

We have a way to upload files which puts them into a `submitted` folder to be reviewed and then moved to the `approved` folder by a site mod/admin. However when we sign in as a mod/admin and try to approve files through the UI we built, it returns the above error on both our local supabase instance in local development as well as a supabase cloud project.

Anyone help is much appreciated! This is the last thing that we need to figure out before releasing to our test server so we are really trying to figure it out quickly!

**Comment by GaryAustin1** [comment]
Update policy normally requires both a "using" and a "with check".  If only using is provided with check will default to that.  Don't think the other way around is true.  
I don't think your select with using will count in place of that.  
But just glancing at what you have quickly.

**Reply by BloxsmithDevelopment** [reply]
Tried that but didn't solve the issue. We think it could be something with us using subfolders.

So we are trying to do a move like `submitted/123/apple.png` to `approved/123/apple.png`. Could this be our issue?

**Reply by GaryAustin1** [reply]
Unless you have some policy that looks at the path for folders then no.   There really is no such thing as a folder, the path is just a name to the actual storage object.  

Maybe you are not formatting the move request correctly?
